File: data/readable_format/cpp/file2435.cpp
#include <iostream>
#include <cstdlib>
#include <cassert>

class MyObj {
public:
    // Overloaded function to simulate optional behavior
    int myfunc() const {
        return 2; // Default behavior when no argument is provided
    }

    int myfunc(int status) const {
        return 1; // Behavior when an argument is provided
    }
};

int main() {
    MyObj myinstance;

    // Test case 1: Call myfunc without arguments
    int res = myinstance.myfunc();
    assert(res == 2 && "Test case 1 failed");

    // Test case 2: Call myfunc with an argument
    res = myinstance.myfunc(1);
    assert(res == 1 && "Test case 2 failed");

    std::cout << "All tests passed." << std::endl;

    return 0;
}

File: data/readable_format/fortran/file2435.f90
module module_myobj

  implicit none

  type :: myobj
  contains
    procedure, nopass :: myfunc
  end type

contains

  integer function myfunc(status)
    integer, optional :: status
    if (present(status)) then
      myfunc = 1
    else
      myfunc = 2
    end if
  end function

end module


program test_myobj

  use :: module_myobj
  implicit none

  integer     :: res = 0
  type(myobj) :: myinstance

  ! Test case 1: Call myfunc without arguments
  res = myinstance%myfunc()
  if (res /= 2) then
    print *, "Test case 1 failed"
    call abort()
  end if

  ! Test case 2: Call myfunc with argument
  res = myinstance%myfunc(1)
  if (res /= 1) then
    print *, "Test case 2 failed"
    call abort()
  end if

  print *, "All tests passed."

end program